File: mro-matlab/lab2/swiss.m
function [ ] = swiss(neighbors)

  data = generate_data('swiss');
%   scatter3(data(:, 1), data(:, 2), data(:, 3));
  
  x = isomap(data, 2, neighbors);
  size(x)
  scatter(x(:, 1), x(:, 2));
end

File: mro-python/src/manifold/swiss.py
from pylab import *
# from mpl_toolkits.mplot3d import Axes3D
from sklearn import manifold, datasets

# load Swiss roll dataset
X, color = datasets.samples_generator.make_swiss_roll(n_samples=1500)

# run Isomap on the points in X with 2 dim output
n_neighbors = 10
Y = manifold.Isomap(n_neighbors, 2).fit_transform(X)

# 3D plot
# fig = figure()
# ax = fig.gca(projection='3d')
# ax.scatter(X[:,0], X[:,1], X[:,2], c=color)

# 2D projection
figure()
scatter(Y[:,0], Y[:,1], c=color)

show()